src/classes/TDTM_Global_API.cls

```diff
@@ -0,0 +1,74 @@
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2016
+* @group TDTM
+* @group-content ../../ApexDocContent/API.htm
+* @description Provides methods to call the Trigger Handler class, which in turns calls the classes that need to
+* run for each DML action on a set of records, and to retrieve the default HEDA TDTM configuration.
+*/
+global class TDTM_Global_API {
+    
+    /*******************************************************************************************************
+    * @description Method to be called from each of the triggers (only one per object) that follow the TDTM 
+    * design. Figures out which classes need to be called, and calls them.
+    * @param isBefore Are we in a before trigger context.
+    * @param isAfter Are we in an after trigger context.
+    * @param isInsert Did the trigger run in response to an insert operation.
+    * @param isUpdate Did the trigger run in response to an update operation.
+    * @param isDelete Did the trigger run in response to a delete operation.
+    * @param isUnDelete Did the trigger run in response to an undelete operation.
+    * @param newList The records that were passed to the trigger as trigger.new.
+    * @param oldList The records that were passed to the trigger as trigger.old.
+    * @param describeObj The type of SObject the class runs for.
+    * @param dao The class that is going to retrieve all the TDTM records.
+    * @return void
+    */
+    global static void run(Boolean isBefore, Boolean isAfter, Boolean isInsert, Boolean isUpdate,
+    Boolean isDelete, Boolean isUnDelete, List<SObject> newList, List<SObject> oldList, 
+    Schema.DescribeSObjectResult describeObj) {
+        
+        TDTM_TriggerHandler.run(isBefore, isAfter, isInsert, isUpdate, isDelete, isUnDelete, newList, oldList, describeObj);
+    }
+    
+     /*******************************************************************************************************
+    * @description Returns the default TDTM configuration. 
+    * @return List<Trigger_Handler__c> a list of trigger handler records with the default values. The records
+    *         have not yet been inserted in the database. You can manipulate them, if necessary, and insert 
+    *         them. If no Trigger_Handler__c records exist, the application will insert the defaults when any 
+    *         of the triggers run. 
+    */
+    global static List<Trigger_Handler__c> getDefaultTdtmConfig() {
+        
```

Comment:
@dospromptman raised a good point here, that we probably shouldn't be returning a specific SObject type. Otherwise are forever tied to it. What if later we want to store the configuration in a custom setting, for example? The challenge is that if we instead return a class we need to make that class and the methods in it global. We need to think about potential alternatives. @kbromer @paulmalenke